tests/trainer/flags/test_min_max_epochs.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+
+from pytorch_lightning import Trainer
+from tests.base.boring_model import BoringModel
+
+
+# @pytest.mark.parametrize("min_epochs", [None, 2])
+# @pytest.mark.parametrize("max_epochs", [None, 3])
+# @pytest.mark.parametrize("min_steps", [None, 20])
+# @pytest.mark.parametrize("max_steps", [None, 100])
```

Comment:
Are all of these 16 configurations valid?
Did you try what happens if we choose max_epochs = 1 and max_steps = 5 but the epoch only has 3 batches?
The number that leads to less training steps should terminate training, right? This could be a separate test.

Reply:
if both are specified, i think should exit out whenever the first condition is met. so for the example you provided, if the epoch has only 3 batches, i agree we should return after the epoch completes and before moving onto the remaining 2 steps. the same goes for min_steps

i think users can already exercise this functionality now, so i'll look to see if there are already test cases for it. if not i'll add them here